### PR TITLE
fix: validation added for empty path and fetched data in withModel()

### PR DESCRIPTION
**Pull request by sharanyavinod**

### CODE REVIEW

**Comment by niekraaijmakers** [comment]
Can you add unit test coverage as well? Looks good to me.

**Comment by sharanyavinod** [comment]
@niekraaijmakers Thanks for the review. I've added changes to improve test coverage.

**Comment by niekraaijmakers** [comment]
@pfauchere WDTY? looks good to me.

File: src/components/ModelProvider.tsx [hunk 1]
```diff
@@ -68,29 +68,34 @@ export class ModelProvider extends Component<ModelProviderType> {
 
     public updateData(cqPath?: string) {
         const path = cqPath || this.props.cqPath;
-        ModelManager.getData({path, forceReload: this.props.cqForceReload}).then((data: Model) => {
-            this.setState(Utils.modelToProps(data));
-        });
+        if (path) {
```
Comment (CezCz) [inline src/components/ModelProvider.tsx:71]: Maybe we can invert it

```
if(!path) {
return
}

<rest of the code>
```

wdyt?

File: src/components/ModelProvider.tsx [hunk 2]
```diff
@@ -68,29 +68,34 @@ export class ModelProvider extends Component<ModelProviderType> {
...
-              `${pagePath}/jcr:content/${itemPath}` :
-              pagePath
-          );
-
-          // Normalize path (replace multiple consecutive slashes with a single one).
-          cqPath = normalizePath(cqPath);
-          this.setState({ cqPath });
-      }
-      return cqPath;
+    private getCQPath(): string {
```
Comment (grubyak) [inline src/components/ModelProvider.tsx:81]: that's a helper that should be in `PathUtils`

**Review by CezCz — APPROVED** [review]

**Review by niekraaijmakers — APPROVED** [review]

**Comment by niekraaijmakers** [comment]
who has rights to merge?  @grubyak can you review ?

File: src/Utils.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@
  * governing permissions and limitations under the License.
  */
 
+import { normalize as normalizePath } from 'path';
```
Comment (grubyak) [inline src/Utils.ts:13]: i would not rename that

**Review by grubyak — APPROVED** [review]

**Pull request merged**